### PR TITLE
Add robot model class

### LINKED ISSUE #721: Add robot model class to database
**Describe the new feature you would like to see**

Include a robot model class in database (instead of an enum). Robot model class defines robot model specific characteristics, add pressure limit and battery limit  as new characteristics.  

**Describe the solution you'd like**
NA

**How will this feature affect the current Threat Model?**
NA

### PR DESCRIPTION
**Pull request by GodVenn**
Closes #721

### CODE REVIEW

File: backend/api/Controllers/Models/CreateRobotModelQuery.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        /// <summary>
+        /// The type of robot model
+        /// </summary>
+        public RobotType RobotType { get; set; }
+
+        /// <summary>
+        /// Lower battery warning threshold in percentage
+        /// </summary>
+        public float BatteryWarningThreshold { get; set; }
```
Comment (Christdej) [inline backend/api/Controllers/Models/CreateRobotModelQuery.cs:15]: Should this be optional aswell?

File: frontend/src/components/Contexts/AuthProvider.tsx [hunk 2]
```diff
@@ -28,7 +28,5 @@ export const AuthProvider = (props: Props) => {
         return () => clearInterval(id)
     }, [])
 
-    VerifyToken()
```
Comment (Christdej) [inline frontend/src/components/Contexts/AuthProvider.tsx:31]: Why is this removed?

File: backend/api.test/EventHandlers/TestMissionScheduler.cs [hunk 3]
```diff
@@ -103,20 +100,30 @@ public void Dispose()
...
 
             var cts = new CancellationTokenSource();
 
             // Add Scheduled mission
-            await _missionService.Create(ScheduledMission);
+            await _missionService.Create(mission);
+
+            _robotControllerMock.RobotServiceMock
+                .Setup(service => service.ReadById(mission.Robot.Id))
+                .Returns(async () => mission.Robot);
```
Comment (Christdej) [inline backend/api.test/EventHandlers/TestMissionScheduler.cs:119]: Alot of methods missing await, might wanna ignore warning or add await

File: backend/api/Controllers/RobotModelController.cs [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    [ProducesResponseType(typeof(RobotModel), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<Robot>> UpdateRobotModelById(
+        [FromRoute] string id,
+        [FromBody] UpdateRobotModelQuery robotModelQuery
+    )
```
Comment (Christdej) [inline backend/api/Controllers/RobotModelController.cs:149]: Would it be better if this was a patch endpoint?

File: backend/api/Database/Models/RobotModel.cs [hunk 5]
```diff
@@ -10,4 +22,52 @@ public enum RobotModel
...
+        [Required]
+        //Save robot type as string in database
+        [Column(TypeName = "nvarchar(56)")]
+        public RobotType Type { get; set; }
+
+        /// <summary>
+        /// Lower battery warning threshold in percentage
+        /// </summary>
+        [Required]
+        public float BatteryWarningThreshold { get; set; }
```
Comment (Christdej) [inline backend/api/Database/Models/RobotModel.cs:44]: Do we want this to be required?

**Review by Christdej — CHANGES_REQUESTED** [review]

File: frontend/src/components/Contexts/AuthProvider.tsx (hunk 2 above)
Comment (GodVenn) [reply frontend/src/components/Contexts/AuthProvider.tsx:31]: It creates an error, shouldn't call callback functions in component definition

File: backend/api/Controllers/RobotModelController.cs (hunk 4 above)
Comment (GodVenn) [reply backend/api/Controllers/RobotModelController.cs:149]: A patch would just updating one field, but here we are updating the entire model

File: backend/api.test/EventHandlers/TestMissionScheduler.cs (hunk 3 above)
Comment (GodVenn) [reply backend/api.test/EventHandlers/TestMissionScheduler.cs:120]: Fixed

File: backend/api/Database/Models/RobotModel.cs (hunk 5 above)
Comment (GodVenn) [reply backend/api/Database/Models/RobotModel.cs:44]: I think you might have a point, it does not need to be required

File: backend/api/Controllers/Models/CreateRobotModelQuery.cs (hunk 1 above)
Comment (GodVenn) [reply backend/api/Controllers/Models/CreateRobotModelQuery.cs:15]: Yes, changing to optional

**Review by Christdej — APPROVED** [review]
